game-core/src/main/java/games/strategy/engine/chat/Chat.java

```diff
@@ -44,26 +41,26 @@
   private final Collection<BiConsumer<PlayerName, String>> statusUpdateListeners =
       new ArrayList<>();
 
-  public Chat(final Messengers messengers, final String chatName) {
-    this.localPlayerName = messengers.getLocalNode().getPlayerName();
-    chatTransmitter = new JavaSocketChatTransmitter(this, chatName, messengers);
+  public Chat(final ChatTransmitter chatTransmitter) {
+    this.localPlayerName = chatTransmitter.getLocalPlayerName();
+    this.chatTransmitter = chatTransmitter;
+    chatTransmitter.setChatClient(this);
     sentMessagesHistory = new SentMessagesHistory();
-    chatters = Optional.ofNullable(chatTransmitter.connect()).orElseGet(HashMap::new);
+    chatters = Optional.ofNullable(chatTransmitter.connect()).orElseGet(HashSet::new);
     updateConnections();
   }
 
   private void updateConnections() {
     final List<ChatParticipant> playerNames =
-        chatters.keySet().stream()
+        chatters.stream()
             .sorted(Comparator.comparing(c -> c.getPlayerName().getValue()))
             .collect(Collectors.toList());
```

Comment:
I wonder if it would make sense to turn `chatters` into a `TreeSet` or something and use this custom comparator.
This has multiple benefits:
1. We don't need to sort it every time a connection is updated
2. The lookup below in linear time `O(n)`, could potentially be done in `O(log n)` using [`TreeSet#ceiling(E)`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/TreeSet.html#ceiling(E)) or something. I haven't found a guarantee for this time-complexity but after googling this, the implementation seems to suggest it.

Of course this only works if there aren't multiple `ChatParticipant`s with the same player name.